### ISSUE TITLE
rook-ceph-crash-collector-keyring secret not created for crash reporter

### ISSUE THREAD

**Issue by NicolaiSchmid** [rook/rook#4553]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**

**Expected behavior:**
On a fresh Kubernetes v1.17 cluster I try to setup an example cluster
```
kubectl apply -f common.yaml
kubectl apply -f operator.yaml
kubectl apply -f cluster.yaml
```
My only changes are rook/ceph:v1.2 tag for the operator and the following diff:
```diff
diff --git a/cluster/examples/kubernetes/ceph/cluster.yaml b/cluster/examples/kubernetes/ceph/cluster.yaml
index 460277c..6d04daf 100644
--- a/cluster/examples/kubernetes/ceph/cluster.yaml
+++ b/cluster/examples/kubernetes/ceph/cluster.yaml
@@ -39,7 +39,7 @@ spec:
   # set the amount of mons to be started
   mon:
     count: 3
-    allowMultiplePerNode: false
+    allowMultiplePerNode: true
   # mgr:
     # modules:
     # Several modules should not need to be included in this list. The "dashboard" and "monitoring" modules
@@ -126,8 +126,8 @@ spec:
 #    osd: rook-ceph-osd-priority-class
 #    mgr: rook-ceph-mgr-priority-class
   storage: # cluster level storage configuration and selection
-    useAllNodes: true
-    useAllDevices: true
+    useAllNodes: false
+    useAllDevices: false
     #deviceFilter:
     config:
       # The default and recommended storeType is dynamically set to bluestore for devices and filestore for directories.
@@ -143,6 +143,12 @@ spec:
     #- path: /var/lib/rook
 # Individual nodes and their config can be specified as well, but 'useAllNodes' above must be set to false. Then, only the named
 # nodes below will be used as storage resources.  Each node's 'name' field should match their 'kubernetes.io/hostname' label.
+    nodes:
+      - name: "master1"
+        devices:
+          - name: "sdb"
+          - name: "sdc"
+          - name: "sdd"
 #    nodes:
 #    - name: "172.17.4.101"
 #      directories: # specific directories to use for storage can be specified for each node
```
With version 1.2 the crashcollector was introduced. And this [PR](https://github.com/rook/rook/pull/4423) changed the collector to use its own secret.

The issue is the collector pod can't mount the ceph secret, because it doesn't exist.
```
Events:
  Type     Reason       Age                   From                      Message
  ----     ------       ----                  ----                      -------
  Normal   Scheduled    6m27s                 default-scheduler         Successfully assigned rook-ceph/rook-ceph-crashcollector-master1.wasc.io-7f94c7bcc7-42lg8 to master1.wasc.io
  Warning  FailedMount  4m24s                 kubelet, master1.wasc.io  Unable to attach or mount volumes: unmounted volumes=[rook-ceph-crash-collector-keyring], unattached volumes=[default-token-78jgn rook-ceph-crash-collector-keyring rook-config-override rook-ceph-log rook-ceph-crash]: timed out waiting for the condition
  Warning  FailedMount  2m8s                  kubelet, master1.wasc.io  Unable to attach or mount volumes: unmounted volumes=[rook-ceph-crash-collector-keyring], unattached volumes=[rook-ceph-crash default-token-78jgn rook-ceph-crash-collector-keyring rook-config-override rook-ceph-log]: timed out waiting for the condition
  Warning  FailedMount  15s (x11 over 6m26s)  kubelet, master1.wasc.io  MountVolume.SetUp failed for volume "rook-ceph-crash-collector-keyring" : secret "rook-ceph-crash-collector-keyring" not found
```
And in fact, it's not here:
```
NAME                                         TYPE                                  DATA   AGE
default-token-78jgn                          kubernetes.io/service-account-token   3      7m47s
rook-ceph-admin-keyring                      kubernetes.io/rook                    1      7m10s
rook-ceph-cmd-reporter-token-2rflk           kubernetes.io/service-account-token   3      7m45s
rook-ceph-config                             kubernetes.io/rook                    2      7m13s
rook-ceph-mgr-token-9d9vw                    kubernetes.io/service-account-token   3      7m45s
rook-ceph-mon                                kubernetes.io/rook                    4      7m13s
rook-ceph-mons-keyring                       kubernetes.io/rook                    1      7m11s
rook-ceph-osd-token-z2bc6                    kubernetes.io/service-account-token   3      7m45s
rook-ceph-system-token-xtdcz                 kubernetes.io/service-account-token   3      7m45s
rook-csi-cephfs-plugin-sa-token-vhkrf        kubernetes.io/service-account-token   3      7m43s
rook-csi-cephfs-provisioner-sa-token-52q5l   kubernetes.io/service-account-token   3      7m43s
rook-csi-rbd-plugin-sa-token-mkq9z           kubernetes.io/service-account-token   3      7m42s
rook-csi-rbd-provisioner-sa-token-mztf8      kubernetes.io/service-account-token   3      7m42s
```

**How to reproduce it (minimal and precise):**

**File(s) to submit**:

**Environment**:
* OS (e.g. from /etc/os-release):
```
NAME="Ubuntu"
VERSION="16.04.6 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.6 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial
```
* Kernel (e.g. `uname -a`):
```
Linux master1 4.15.0-72-generic #81~16.04.1-Ubuntu SMP Tue Nov 26 16:34:21 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```
* Cloud provider or hardware configuration:
Bare metal
* Rook version (use `rook version` inside of a Rook Pod):
```
rook: v1.2.0
go: go1.11
```
* Storage backend version (e.g. for ceph do `ceph -v`):
```
ceph version 14.2.5 (ad5bd132e1492173c85fda2cc863152730b16a92) nautilus (stable)
```
* Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"", Minor:"", GitVersion:"v0.0.0-master+$Format:%h$", GitCommit:"$Format:%H$", GitTreeState:"", BuildDate:"1970-01-01T00:00:00Z", GoVersion:"go1.13.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.0", GitCommit:"70132b0f130acc0bed193d9ba59dd186f0e634cf", GitTreeState:"clean", BuildDate:"2019-12-07T21:12:17Z", GoVersion:"go1.13.4", Compiler:"gc", Platform:"linux/amd64"}
```
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift):
baremetal with kubeadm
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/master/ceph-toolbox.html)):
The cluster is not created yet

**Comment by leseb** [comment]
This should happen after the mons orchestration sequence, can you restart the operator and share the logs?
Typically, the crash pod should hang a little, waiting for the volume, but this shouldn't take much time to reconcile.

Thanks.

**Comment by NicolaiSchmid** [comment]
@BlaineEXE 
```
NAME                                                        READY   STATUS     RESTARTS   AGE
csi-cephfsplugin-8scwh                                      3/3     Running    0          11h
csi-cephfsplugin-provisioner-56c8b7ddf4-m5qbg               4/4     Running    0          11h
csi-cephfsplugin-provisioner-56c8b7ddf4-mrf2w               4/4     Running    0          11h
csi-rbdplugin-npm2m                                         3/3     Running    0          11h
csi-rbdplugin-provisioner-6ff4dd4b94-2qw44                  5/5     Running    0          11h
csi-rbdplugin-provisioner-6ff4dd4b94-rtxlx                  5/5     Running    0          11h
rook-ceph-crashcollector-master1.wasc.io-6c79cd5468-5ww9f   0/1     Init:0/2   0          11h
rook-ceph-mon-a-566b9665c8-g26c7                            1/1     Running    0          11h
rook-ceph-mon-d-676c468989-fwkxh                            1/1     Running    0          11h
rook-ceph-mon-f-7478fd675-hxvzj                             1/1     Running    0          10h
rook-ceph-operator-56654965c7-n2pxp                         1/1     Running    0          11h
rook-ceph-tools-7f96779fb9-wmtxx                            1/1     Running    0          11h
rook-discover-shltw                                         1/1     Running    0          11h
```
Operator logs:
```
2019-12-19 22:01:59.966535 I | rookcmd: starting Rook v1.2.0 with arguments '/usr/local/bin/rook ceph operator'
2019-12-19 22:01:59.966631 I | rookcmd: flag values: --add_dir_header=false, --alsologtostderr=false, --csi-attacher-image=quay.io/k8scsi/csi-attacher:v1.2.0, --csi-ceph-image=quay.io/cephcsi/cephcsi:v1.2.2, --csi-cephfs-plugin-template-path=/etc/ceph-csi/cephfs/csi-cephfsplugin.yaml, --csi-cephfs-provisioner-dep-template-path=/etc/ceph-csi/cephfs/csi-cephfsplugin-provisioner-dep.yaml, --csi-cephfs-provisioner-sts-template-path=/etc/ceph-csi/cephfs/csi-cephfsplugin-provisioner-sts.yaml, --csi-driver-name-prefix=, --csi-enable-cephfs=true, --csi-enable-grpc-metrics=true, --csi-enable-rbd=true, --csi-kubelet-dir-path=/var/lib/kubelet, --csi-provisioner-image=quay.io/k8scsi/csi-provisioner:v1.4.0, --csi-rbd-plugin-template-path=/etc/ceph-csi/rbd/csi-rbdplugin.yaml, --csi-rbd-provisioner-dep-template-path=/etc/ceph-csi/rbd/csi-rbdplugin-provisioner-dep.yaml, --csi-rbd-provisioner-sts-template-path=/etc/ceph-csi/rbd/csi-rbdplugin-provisioner-sts.yaml, --csi-registrar-image=quay.io/k8scsi/csi-node-driver-registrar:v1.1.0, --csi-snapshotter-image=quay.io/k8scsi/csi-snapshotter:v1.2.2, --enable-discovery-daemon=true, --enable-flex-driver=false, --enable-machine-disruption-budget=false, --help=false, --kubeconfig=, --log-flush-frequency=5s, --log-level=INFO, --log_backtrace_at=:0, --log_dir=, --log_file=, --log_file_max_size=1800, --logtostderr=true, --master=, --mon-healthcheck-interval=45s, --mon-out-timeout=10m0s, --operator-image=, --service-account=, --skip_headers=false, --skip_log_headers=false, --stderrthreshold=2, --v=0, --vmodule=
2019-12-19 22:01:59.966636 I | cephcmd: starting operator
2019-12-19 22:02:00.002193 I | op-discover: rook-discover daemonset started
2019-12-19 22:02:00.003806 I | operator: rook-provisioner rook.io/block started using rook.io flex vendor dir
I1219 22:02:00.004098       8 leaderelection.go:217] attempting to acquire leader lease  rook-ceph/rook.io-block...
2019-12-19 22:02:00.004648 I | operator: rook-provisioner ceph.rook.io/block started using ceph.rook.io flex vendor dir
2019-12-19 22:02:00.004666 I | operator: Watching all namespaces for cluster CRDs
2019-12-19 22:02:00.004682 I | op-cluster: start watching clusters in all namespaces
2019-12-19 22:02:00.004810 I | op-cluster: Enabling hotplug orchestration: ROOK_DISABLE_DEVICE_HOTPLUG=false
2019-12-19 22:02:00.004955 I | operator: setting up the controller-runtime manager
I1219 22:02:00.004993       8 leaderelection.go:217] attempting to acquire leader lease  rook-ceph/ceph.rook.io-block...
2019-12-19 22:02:00.020564 I | op-cluster: starting cluster in namespace rook-ceph
I1219 22:02:00.022813       8 leaderelection.go:227] successfully acquired lease rook-ceph/ceph.rook.io-block
I1219 22:02:00.022892       8 controller.go:769] Starting provisioner controller ceph.rook.io/block_rook-ceph-operator-56654965c7-n2pxp_2e5116d7-22ab-11ea-a088-622e32bc74cd!
I1219 22:02:00.023427       8 event.go:209] Event(v1.ObjectReference{Kind:"Endpoints", Namespace:"rook-ceph", Name:"ceph.rook.io-block", UID:"41797015-7e25-4775-a3ba-56f932e9bfc7", APIVersion:"v1", ResourceVersion:"84336", FieldPath:""}): type: 'Normal' reason: 'LeaderElection' rook-ceph-operator-56654965c7-n2pxp_2e5116d7-22ab-11ea-a088-622e32bc74cd became leader
I1219 22:02:00.024260       8 leaderelection.go:227] successfully acquired lease rook-ceph/rook.io-block
I1219 22:02:00.024335       8 controller.go:769] Starting provisioner controller rook.io/block_rook-ceph-operator-56654965c7-n2pxp_2e50f263-22ab-11ea-a088-622e32bc74cd!
I1219 22:02:00.024347       8 event.go:209] Event(v1.ObjectReference{Kind:"Endpoints", Namespace:"rook-ceph", Name:"rook.io-block", UID:"1056af66-ca35-451f-8b7c-4fee8be0cd46", APIVersion:"v1", ResourceVersion:"84335", FieldPath:""}): type: 'Normal' reason: 'LeaderElection' rook-ceph-operator-56654965c7-n2pxp_2e50f263-22ab-11ea-a088-622e32bc74cd became leader
2019-12-19 22:02:00.514915 I | operator: starting the controller-runtime manager
I1219 22:02:00.623065       8 controller.go:818] Started provisioner controller ceph.rook.io/block_rook-ceph-operator-56654965c7-n2pxp_2e5116d7-22ab-11ea-a088-622e32bc74cd!
I1219 22:02:01.624551       8 controller.go:818] Started provisioner controller rook.io/block_rook-ceph-operator-56654965c7-n2pxp_2e50f263-22ab-11ea-a088-622e32bc74cd!
2019-12-19 22:02:02.400572 I | ceph-csi: CSIDriver object created for driver "rook-ceph.rbd.csi.ceph.com"
2019-12-19 22:02:02.407662 I | ceph-csi: CSIDriver object created for driver "rook-ceph.cephfs.csi.ceph.com"
2019-12-19 22:02:02.407687 I | operator: successfully started Ceph CSI driver(s)
2019-12-19 22:02:08.407831 I | op-cluster: detecting the ceph image version for image ceph/ceph:v14.2.5-20191210...
2019-12-19 22:02:11.646581 I | op-cluster: Detected ceph image version: "14.2.5-0 nautilus"
2019-12-19 22:02:11.647696 E | cephconfig: clusterInfo: <nil>
2019-12-19 22:02:11.647721 I | op-cluster: CephCluster "rook-ceph" status: "Creating". 
2019-12-19 22:02:11.656928 I | op-mon: start running mons
2019-12-19 22:02:11.658124 I | exec: Running command: ceph-authtool --create-keyring /var/lib/rook/rook-ceph/mon.keyring --gen-key -n mon. --cap mon 'allow *'
2019-12-19 22:02:11.712919 I | exec: Running command: ceph-authtool --create-keyring /var/lib/rook/rook-ceph/client.admin.keyring --gen-key -n client.admin --cap mon 'allow *' --cap osd 'allow *' --cap mgr 'allow *' --cap mds 'allow'
2019-12-19 22:02:11.772419 I | op-mon: creating mon secrets for a new cluster
2019-12-19 22:02:11.781229 I | op-mon: saved mon endpoints to config map map[data: maxMonId:-1 mapping:{"node":{}} csi-cluster-config-json:[{"clusterID":"rook-ceph","monitors":[]}]]
2019-12-19 22:02:11.843370 I | cephconfig: writing config file /var/lib/rook/rook-ceph/rook-ceph.config
2019-12-19 22:02:11.843475 I | cephconfig: generated admin config in /var/lib/rook/rook-ceph
2019-12-19 22:02:13.843018 I | op-mon: targeting the mon count 3
2019-12-19 22:02:13.845638 I | op-mon: sched-mon: created canary deployment rook-ceph-mon-a-canary
2019-12-19 22:02:14.271608 I | op-mon: sched-mon: canary monitor deployment rook-ceph-mon-a-canary scheduled to master1.wasc.io
2019-12-19 22:02:14.271626 I | op-mon: assignmon: mon a assigned to node master1.wasc.io
2019-12-19 22:02:14.274621 I | op-mon: sched-mon: created canary deployment rook-ceph-mon-b-canary
2019-12-19 22:02:14.642821 I | op-mon: sched-mon: canary monitor deployment rook-ceph-mon-b-canary scheduled to master1.wasc.io
2019-12-19 22:02:14.642839 I | op-mon: assignmon: mon b assigned to node master1.wasc.io
2019-12-19 22:02:14.645299 I | op-mon: sched-mon: created canary deployment rook-ceph-mon-c-canary
2019-12-19 22:02:15.242805 I | op-mon: sched-mon: canary monitor deployment rook-ceph-mon-c-canary scheduled to master1.wasc.io
2019-12-19 22:02:15.242825 I | op-mon: assignmon: mon c assigned to node master1.wasc.io
2019-12-19 22:02:15.242834 I | op-mon: assignmon: cleaning up canary deployment rook-ceph-mon-a-canary and canary pvc 
2019-12-19 22:02:15.242842 I | op-k8sutil: removing deployment rook-ceph-mon-a-canary if it exists
2019-12-19 22:02:15.245152 I | op-k8sutil: Removed deployment rook-ceph-mon-a-canary
2019-12-19 22:02:15.247542 I | op-k8sutil: rook-ceph-mon-a-canary still found. waiting...
2019-12-19 22:02:17.249956 I | op-k8sutil: rook-ceph-mon-a-canary still found. waiting...
2019-12-19 22:02:19.252097 I | op-k8sutil: rook-ceph-mon-a-canary still found. waiting...
2019-12-19 22:02:21.254505 I | op-k8sutil: rook-ceph-mon-a-canary still found. waiting...
2019-12-19 22:02:23.256591 I | op-k8sutil: rook-ceph-mon-a-canary still found. waiting...
2019-12-19 22:02:25.258984 I | op-k8sutil: rook-ceph-mon-a-canary still found. waiting...
2019-12-19 22:02:27.261444 I | op-k8sutil: rook-ceph-mon-a-canary still found. waiting...
2019-12-19 22:02:29.263807 I | op-k8sutil: rook-ceph-mon-a-canary still found. waiting...
2019-12-19 22:02:31.266489 I | op-k8sutil: rook-ceph-mon-a-canary still found. waiting...
2019-12-19 22:02:33.268792 I | op-k8sutil: rook-ceph-mon-a-canary still found. waiting...
2019-12-19 22:02:35.271422 I | op-k8sutil: rook-ceph-mon-a-canary still found. waiting...
2019-12-19 22:02:37.273771 I | op-k8sutil: rook-ceph-mon-a-canary still found. waiting...
2019-12-19 22:02:39.276066 I | op-k8sutil: rook-ceph-mon-a-canary still found. waiting...
2019-12-19 22:02:41.278930 I | op-k8sutil: rook-ceph-mon-a-canary still found. waiting...
2019-12-19 22:02:43.281273 I | op-k8sutil: rook-ceph-mon-a-canary still found. waiting...
2019-12-19 22:02:45.283500 I | op-k8sutil: rook-ceph-mon-a-canary still found. waiting...
2019-12-19 22:02:47.285350 I | op-k8sutil: rook-ceph-mon-a-canary still found. waiting...
2019-12-19 22:02:49.287629 I | op-k8sutil: rook-ceph-mon-a-canary still found. waiting...
2019-12-19 22:02:51.290270 I | op-k8sutil: rook-ceph-mon-a-canary still found. waiting...
2019-12-19 22:02:53.292590 I | op-k8sutil: rook-ceph-mon-a-canary still found. waiting...
2019-12-19 22:02:55.294756 I | op-k8sutil: rook-ceph-mon-a-canary still found. waiting...
2019-12-19 22:02:57.297131 I | op-k8sutil: rook-ceph-mon-a-canary still found. waiting...
2019-12-19 22:02:59.299246 I | op-k8sutil: rook-ceph-mon-a-canary still found. waiting...
2019-12-19 22:03:01.303353 I | op-k8sutil: rook-ceph-mon-a-canary still found. waiting...
2019-12-19 22:03:03.306971 I | op-k8sutil: rook-ceph-mon-a-canary still found. waiting...
2019-12-19 22:03:05.310432 I | op-k8sutil: rook-ceph-mon-a-canary still found. waiting...
2019-12-19 22:03:07.313884 I | op-k8sutil: rook-ceph-mon-a-canary still found. waiting...
2019-12-19 22:03:09.317373 I | op-k8sutil: rook-ceph-mon-a-canary still found. waiting...
2019-12-19 22:03:11.321389 I | op-k8sutil: rook-ceph-mon-a-canary still found. waiting...
2019-12-19 22:03:13.324424 I | op-k8sutil: rook-ceph-mon-a-canary still found. waiting...
2019-12-19 22:03:15.324605 I | op-mon: assignmon: error deleting canary deployment rook-ceph-mon-a-canary. gave up waiting for rook-ceph-mon-a-canary pods to be terminated
2019-12-19 22:03:15.324640 I | op-mon: assignmon: cleaning up canary deployment rook-ceph-mon-b-canary and canary pvc 
2019-12-19 22:03:15.324654 I | op-k8sutil: removing deployment rook-ceph-mon-b-canary if it exists
2019-12-19 22:03:15.332120 I | op-k8sutil: Removed deployment rook-ceph-mon-b-canary
2019-12-19 22:03:15.335303 I | op-k8sutil: rook-ceph-mon-b-canary still found. waiting...
2019-12-19 22:03:17.338652 I | op-k8sutil: rook-ceph-mon-b-canary still found. waiting...
2019-12-19 22:03:19.342094 I | op-k8sutil: rook-ceph-mon-b-canary still found. waiting...
2019-12-19 22:03:21.344848 I | op-k8sutil: rook-ceph-mon-b-canary still found. waiting...
2019-12-19 22:03:23.347495 I | op-k8sutil: rook-ceph-mon-b-canary still found. waiting...
2019-12-19 22:03:25.350309 I | op-k8sutil: confirmed rook-ceph-mon-b-canary does not exist
2019-12-19 22:03:25.350338 I | op-mon: assignmon: cleaning up canary deployment rook-ceph-mon-c-canary and canary pvc 
2019-12-19 22:03:25.350349 I | op-k8sutil: removing deployment rook-ceph-mon-c-canary if it exists
2019-12-19 22:03:25.355133 I | op-k8sutil: Removed deployment rook-ceph-mon-c-canary
2019-12-19 22:03:25.357289 I | op-k8sutil: rook-ceph-mon-c-canary still found. waiting...
2019-12-19 22:03:27.360679 I | op-k8sutil: rook-ceph-mon-c-canary still found. waiting...
2019-12-19 22:03:29.378017 I | op-k8sutil: confirmed rook-ceph-mon-c-canary does not exist
2019-12-19 22:03:29.378036 I | op-mon: creating mon a
2019-12-19 22:03:29.384037 I | op-mon: mon "a" endpoint are [v2:10.96.81.163:3300,v1:10.96.81.163:6789]
2019-12-19 22:03:29.389832 I | op-mon: saved mon endpoints to config map map[data:a=10.96.81.163:6789 maxMonId:2 mapping:{"node":{"a":{"Name":"master1.wasc.io","Hostname":"master1.wasc.io","Address":"138.201.216.36"},"b":{"Name":"master1.wasc.io","Hostname":"master1.wasc.io","Address":"138.201.216.36"},"c":{"Name":"master1.wasc.io","Hostname":"master1.wasc.io","Address":"138.201.216.36"}}} csi-cluster-config-json:[{"clusterID":"rook-ceph","monitors":["10.96.81.163:6789"]}]]
2019-12-19 22:03:29.393610 I | cephconfig: writing config file /var/lib/rook/rook-ceph/rook-ceph.config
2019-12-19 22:03:29.393734 I | cephconfig: generated admin config in /var/lib/rook/rook-ceph
2019-12-19 22:03:29.404982 I | cephconfig: writing config file /var/lib/rook/rook-ceph/rook-ceph.config
2019-12-19 22:03:29.405072 I | cephconfig: generated admin config in /var/lib/rook/rook-ceph
2019-12-19 22:03:29.407078 I | op-mon: 0 of 1 expected mon deployments exist. creating new deployment(s).
2019-12-19 22:03:29.412163 I | op-mon: waiting for mon quorum with [a]
2019-12-19 22:03:29.420258 I | op-mon: mon a is not yet running
2019-12-19 22:03:29.420284 I | op-mon: mons running: []
2019-12-19 22:03:29.420493 I | exec: Running command: ceph quorum_status --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/209553242
2019-12-19 22:03:29.506934 E | ceph-crashcollector-controller: node reconcile failed on op "unchanged": Operation cannot be fulfilled on deployments.apps "rook-ceph-crashcollector-master1.wasc.io": the object has been modified; please apply your changes to the latest version and try again
2019-12-19 22:03:44.541775 I | exec: timed out
2019-12-19 22:03:49.547241 I | op-mon: mons running: [a]
2019-12-19 22:03:49.547408 I | exec: Running command: ceph quorum_status --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/114035441
2019-12-19 22:04:04.657789 I | exec: timed out
```
After that it just retries the ceph command but fails over and over



@leseb 
I restarted the operator and these are the logs:
```
2019-12-20 09:18:27.499810 I | rookcmd: starting Rook v1.2.0 with arguments '/usr/local/bin/rook ceph operator'
2019-12-20 09:18:27.499900 I | rookcmd: flag values: --add_dir_header=false, --alsologtostderr=false, --csi-attacher-image=quay.io/k8scsi/csi-attacher:v1.2.0, --csi-ceph-image=quay.io/cephcsi/cephcsi:v1.2.2, --csi-cephfs-plugin-template-path=/etc/ceph-csi/cephfs/csi-cephfsplugin.yaml, --csi-cephfs-provisioner-dep-template-path=/etc/ceph-csi/cephfs/csi-cephfsplugin-provisioner-dep.yaml, --csi-cephfs-provisioner-sts-template-path=/etc/ceph-csi/cephfs/csi-cephfsplugin-provisioner-sts.yaml, --csi-driver-name-prefix=, --csi-enable-cephfs=true, --csi-enable-grpc-metrics=true, --csi-enable-rbd=true, --csi-kubelet-dir-path=/var/lib/kubelet, --csi-provisioner-image=quay.io/k8scsi/csi-provisioner:v1.4.0, --csi-rbd-plugin-template-path=/etc/ceph-csi/rbd/csi-rbdplugin.yaml, --csi-rbd-provisioner-dep-template-path=/etc/ceph-csi/rbd/csi-rbdplugin-provisioner-dep.yaml, --csi-rbd-provisioner-sts-template-path=/etc/ceph-csi/rbd/csi-rbdplugin-provisioner-sts.yaml, --csi-registrar-image=quay.io/k8scsi/csi-node-driver-registrar:v1.1.0, --csi-snapshotter-image=quay.io/k8scsi/csi-snapshotter:v1.2.2, --enable-discovery-daemon=true, --enable-flex-driver=false, --enable-machine-disruption-budget=false, --help=false, --kubeconfig=, --log-flush-frequency=5s, --log-level=INFO, --log_backtrace_at=:0, --log_dir=, --log_file=, --log_file_max_size=1800, --logtostderr=true, --master=, --mon-healthcheck-interval=45s, --mon-out-timeout=10m0s, --operator-image=, --service-account=, --skip_headers=false, --skip_log_headers=false, --stderrthreshold=2, --v=0, --vmodule=
2019-12-20 09:18:27.499905 I | cephcmd: starting operator
2019-12-20 09:18:27.522939 I | op-discover: rook-discover daemonset already exists, updating ...
2019-12-20 09:18:27.530844 I | operator: rook-provisioner ceph.rook.io/block started using ceph.rook.io flex vendor dir
I1220 09:18:27.530945       8 leaderelection.go:217] attempting to acquire leader lease  rook-ceph/ceph.rook.io-block...
2019-12-20 09:18:27.531324 I | operator: rook-provisioner rook.io/block started using rook.io flex vendor dir
2019-12-20 09:18:27.531334 I | operator: Watching all namespaces for cluster CRDs
2019-12-20 09:18:27.531362 I | op-cluster: start watching clusters in all namespaces
I1220 09:18:27.531398       8 leaderelection.go:217] attempting to acquire leader lease  rook-ceph/rook.io-block...
2019-12-20 09:18:27.531385 I | operator: setting up the controller-runtime manager
2019-12-20 09:18:27.531578 I | op-cluster: Enabling hotplug orchestration: ROOK_DISABLE_DEVICE_HOTPLUG=false
2019-12-20 09:18:27.540717 I | op-cluster: starting cluster in namespace rook-ceph
2019-12-20 09:18:28.050853 I | operator: starting the controller-runtime manager
2019-12-20 09:18:28.509188 I | ceph-csi: CSIDriver CRD already had been registered for "rook-ceph.rbd.csi.ceph.com"
2019-12-20 09:18:28.512117 I | ceph-csi: CSIDriver CRD already had been registered for "rook-ceph.cephfs.csi.ceph.com"
2019-12-20 09:18:28.512142 I | operator: successfully started Ceph CSI driver(s)
2019-12-20 09:18:34.512392 I | op-cluster: detecting the ceph image version for image ceph/ceph:v14.2.5-20191210...
2019-12-20 09:18:36.988950 I | op-cluster: Detected ceph image version: "14.2.5-0 nautilus"
2019-12-20 09:18:36.993298 I | op-mon: parsing mon endpoints: f=10.96.201.55:6789,a=10.96.81.163:6789,d=10.96.175.80:6789
2019-12-20 09:18:36.993416 I | op-mon: loaded: maxMonID=5, mons=map[f:0xc000d5c8a0 a:0xc000d5c8e0 d:0xc000d5c920], mapping=&{Node:map[c:0xc000abf470 d:0xc000abf4a0 e:0xc000abf4d0 f:0xc000abf500 a:0xc000abf3b0 b:0xc000abf440]}
2019-12-20 09:18:36.993835 I | cephconfig: writing config file /var/lib/rook/rook-ceph/rook-ceph.config
2019-12-20 09:18:36.993956 I | cephconfig: generated admin config in /var/lib/rook/rook-ceph
2019-12-20 09:18:36.994089 I | exec: Running command: ceph versions --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/120258047
I1220 09:18:44.443193       8 leaderelection.go:227] successfully acquired lease rook-ceph/rook.io-block
I1220 09:18:44.443238       8 event.go:209] Event(v1.ObjectReference{Kind:"Endpoints", Namespace:"rook-ceph", Name:"rook.io-block", UID:"1056af66-ca35-451f-8b7c-4fee8be0cd46", APIVersion:"v1", ResourceVersion:"257008", FieldPath:""}): type: 'Normal' reason: 'LeaderElection' rook-ceph-operator-56654965c7-hrzdp_ae5e344b-2309-11ea-9279-0a58816d88d5 became leader
I1220 09:18:44.443299       8 controller.go:769] Starting provisioner controller rook.io/block_rook-ceph-operator-56654965c7-hrzdp_ae5e344b-2309-11ea-9279-0a58816d88d5!
I1220 09:18:44.543540       8 controller.go:818] Started provisioner controller rook.io/block_rook-ceph-operator-56654965c7-hrzdp_ae5e344b-2309-11ea-9279-0a58816d88d5!
I1220 09:18:46.087349       8 leaderelection.go:227] successfully acquired lease rook-ceph/ceph.rook.io-block
I1220 09:18:46.087450       8 controller.go:769] Starting provisioner controller ceph.rook.io/block_rook-ceph-operator-56654965c7-hrzdp_ae5e2092-2309-11ea-9279-0a58816d88d5!
I1220 09:18:46.087496       8 event.go:209] Event(v1.ObjectReference{Kind:"Endpoints", Namespace:"rook-ceph", Name:"ceph.rook.io-block", UID:"41797015-7e25-4775-a3ba-56f932e9bfc7", APIVersion:"v1", ResourceVersion:"257013", FieldPath:""}): type: 'Normal' reason: 'LeaderElection' rook-ceph-operator-56654965c7-hrzdp_ae5e2092-2309-11ea-9279-0a58816d88d5 became leader
I1220 09:18:46.187710       8 controller.go:818] Started provisioner controller ceph.rook.io/block_rook-ceph-operator-56654965c7-hrzdp_ae5e2092-2309-11ea-9279-0a58816d88d5!
2019-12-20 09:18:52.105245 I | exec: timed out
2019-12-20 09:18:52.105383 E | op-cluster: failed to get ceph daemons versions. failed to run 'ceph versions: exit status 1
2019-12-20 09:18:52.105404 I | op-cluster: CephCluster "rook-ceph" status: "Creating". 
2019-12-20 09:18:52.121850 I | op-mon: start running mons
2019-12-20 09:18:52.125189 I | op-mon: parsing mon endpoints: f=10.96.201.55:6789,a=10.96.81.163:6789,d=10.96.175.80:6789
2019-12-20 09:18:52.125270 I | op-mon: loaded: maxMonID=5, mons=map[f:0xc000e56580 a:0xc000e565c0 d:0xc000e56600], mapping=&{Node:map[a:0xc000e4a540 b:0xc000e4a570 c:0xc000e4a5a0 d:0xc000e4a5d0 e:0xc000e4a600 f:0xc000e4a630]}
2019-12-20 09:18:52.129575 I | op-mon: saved mon endpoints to config map map[maxMonId:5 mapping:{"node":{"a":{"Name":"master1.wasc.io","Hostname":"master1.wasc.io","Address":"138.201.216.36"},"b":{"Name":"master1.wasc.io","Hostname":"master1.wasc.io","Address":"138.201.216.36"},"c":{"Name":"master1.wasc.io","Hostname":"master1.wasc.io","Address":"138.201.216.36"},"d":{"Name":"master1.wasc.io","Hostname":"master1.wasc.io","Address":"138.201.216.36"},"e":{"Name":"master1.wasc.io","Hostname":"master1.wasc.io","Address":"138.201.216.36"},"f":{"Name":"master1.wasc.io","Hostname":"master1.wasc.io","Address":"138.201.216.36"}}} csi-cluster-config-json:[{"clusterID":"rook-ceph","monitors":["10.96.201.55:6789","10.96.81.163:6789","10.96.175.80:6789"]}] data:f=10.96.201.55:6789,a=10.96.81.163:6789,d=10.96.175.80:6789]
2019-12-20 09:18:52.133909 I | cephconfig: writing config file /var/lib/rook/rook-ceph/rook-ceph.config
2019-12-20 09:18:52.134002 I | cephconfig: generated admin config in /var/lib/rook/rook-ceph
2019-12-20 09:18:53.508647 I | op-mon: targeting the mon count 3
2019-12-20 09:18:53.508820 I | exec: Running command: ceph config set global mon_allow_pool_delete true --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/293789522
2019-12-20 09:19:08.626974 I | exec: timed out
2019-12-20 09:19:08.627178 W | op-mon: failed to set Rook and/or user-defined Ceph config options before starting mons; will retry after starting mons. failed to apply default Ceph configurations: failed to set one or more Ceph configs: failed to set ceph config in the centralized mon configuration database; you may need to use the rook-config-override ConfigMap. output: timed out
: exit status 1
2019-12-20 09:19:08.627204 I | op-mon: checking for basic quorum with existing mons
2019-12-20 09:19:08.640283 I | op-mon: mon "f" endpoint are [v2:10.96.201.55:3300,v1:10.96.201.55:6789]
2019-12-20 09:19:08.649662 I | op-mon: mon "a" endpoint are [v2:10.96.81.163:3300,v1:10.96.81.163:6789]
2019-12-20 09:19:08.658312 I | op-mon: mon "d" endpoint are [v2:10.96.175.80:3300,v1:10.96.175.80:6789]
2019-12-20 09:19:08.830666 I | op-mon: saved mon endpoints to config map map[data:f=10.96.201.55:6789,a=10.96.81.163:6789,d=10.96.175.80:6789 maxMonId:5 mapping:{"node":{"a":{"Name":"master1.wasc.io","Hostname":"master1.wasc.io","Address":"138.201.216.36"},"b":{"Name":"master1.wasc.io","Hostname":"master1.wasc.io","Address":"138.201.216.36"},"c":{"Name":"master1.wasc.io","Hostname":"master1.wasc.io","Address":"138.201.216.36"},"d":{"Name":"master1.wasc.io","Hostname":"master1.wasc.io","Address":"138.201.216.36"},"e":{"Name":"master1.wasc.io","Hostname":"master1.wasc.io","Address":"138.201.216.36"},"f":{"Name":"master1.wasc.io","Hostname":"master1.wasc.io","Address":"138.201.216.36"}}} csi-cluster-config-json:[{"clusterID":"rook-ceph","monitors":["10.96.175.80:6789","10.96.201.55:6789","10.96.81.163:6789"]}]]
2019-12-20 09:19:09.231164 I | cephconfig: writing config file /var/lib/rook/rook-ceph/rook-ceph.config
2019-12-20 09:19:09.231327 I | cephconfig: generated admin config in /var/lib/rook/rook-ceph
2019-12-20 09:19:10.030784 I | cephconfig: writing config file /var/lib/rook/rook-ceph/rook-ceph.config
2019-12-20 09:19:10.030992 I | cephconfig: generated admin config in /var/lib/rook/rook-ceph
2019-12-20 09:19:10.038359 I | op-mon: deployment for mon rook-ceph-mon-f already exists. updating if needed
2019-12-20 09:19:10.040321 I | op-k8sutil: updating deployment rook-ceph-mon-f
2019-12-20 09:19:12.049777 I | op-k8sutil: finished waiting for updated deployment rook-ceph-mon-f
2019-12-20 09:19:12.049805 I | op-mon: waiting for mon quorum with [f a d]
2019-12-20 09:19:12.063890 I | op-mon: mons running: [f a d]
2019-12-20 09:19:12.064018 I | exec: Running command: ceph quorum_status --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/005657225
2019-12-20 09:19:27.180290 I | exec: timed out
2019-12-20 09:19:32.197306 I | op-mon: mons running: [f a d]
2019-12-20 09:19:32.197416 I | exec: Running command: ceph quorum_status --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/786102356
2019-12-20 09:19:47.298425 I | exec: timed out
2019-12-20 09:19:52.316358 I | op-mon: mons running: [f a d]
2019-12-20 09:19:52.316472 I | exec: Running command: ceph quorum_status --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/315995555
```
Again the crashcollector pod wants to mount the secret `rook-ceph-crash-collector-keyring`, but it doesn't exist in the namespace

**Comment by leseb** [comment]
@NicolaiSchmid well it looks like you're having a different problem here, are the mons in quorum? It seems that the operator fails to run any Ceph command.

**Comment by NicolaiSchmid** [comment]
As usual, I forgot to zap my disks and the `/var/lib/rook directory` before recreating the cluster!
I had encryption enabled before, and the mons freaked out about it of course!
Thanks to everyone and their help!

**Comment by kookone** [comment]
I also encountered this problem，I tried but it still doesn't work

> As usual, I forgot to zap my disks and the `/var/lib/rook directory` before recreating the cluster!
> I had encryption enabled before, and the mons freaked out about it of course!
> Thanks to everyone and their help!

**Comment by nielsbasjes** [comment]
I also have this problem when trying to install everything on a set of old servergrade systems.

I also have the 

     rook-ceph-crashcollector-node8-f549c46d6-4dtgs

fail with 

    MountVolume.SetUp failed for volume "rook-ceph-crash-collector-keyring" : secret "rook-ceph-crash-collector-keyring" not found

At this moment I seem to see the pattern that this ONLY happens on the node where the ```rook-ceph-mon-a-8686d87dd9-9q99t``` is also running.

Perhaps these two are conflicting each other at the local disk level?

**Comment by nielsbasjes** [comment]
I think I figured it out. Writing this down here for others to find.
I had done several tests installing rook ceph in Kubernetes.
Cleaning a system for a new run takes more than just removing the /var/lib/rook directory.
I also found some sockets in the /var/lib/kubelet directory related to rook ceph
On my system it worked after removing these too.

**Comment by secops360** [comment]
It is not quorum, operator pod can't access (port 6789 for ceph mon) for some reason. (try it with curl -v inside operator pod). I've moved operator pod to the same node with mon node and it worked. Deleting  sockets in /var/lib/kubelet/plugin didn't help, @nielsbasjes most likely you had operator pod and mon pod sitting on same node after your cleanup. Can't explain why can't talk to mon on other k8s node yet though, will keep you posted.

**Comment by harryleesan** [comment]
This bug seems to appear in Rook 1.2 when connecting to an **external** Ceph cluster.

After creating the CRs: `CephCluster` (external=true) -> `CephFilesystem`, the `rook-ceph-crash-collector-keyring` secret is not created and crash reporter fails to start.

I had to create the rook-ceph-crash-collector-keyring manually.

**Comment by BlaineEXE** [comment]
@leseb is this something you've seen in your testing or something you have tested? 

@harryleesan if you look at the operator logs, do you see any permissions errors like `nodes ... is forbidden: User ... cannot get resource in API group ...`?

**Comment by harryleesan** [comment]
@BlaineEXE I don’t see any errors in the operator logs.

**Extra info:**
After creating **CephCluster** (external=true), I had to manually create `rook-config-override` configmap to create **CephFilesystem**.

```yaml
apiVersion: ceph.rook.io/v1                                                                                                                         
kind: CephCluster                                                                                                                                   
metadata:                                                                                                                                           
    name: rook-ceph-external                                                                                               
    namespace: rook-ceph                                                                                                                 
spec:                                                                                                                                               
    external:                                                                                                                                         
        enable: true                                                                                                                                    
    dataDirHostPath: /var/lib/rook                                                                                                                    
    cephVersion: 
         image: ceph/ceph:v14.2.5
```

```yaml
  apiVersion: ceph.rook.io/v1                                                                                                                             
  kind: CephFilesystem                                                                                                                                    
  metadata:                                                                                                                                               
     name: myfs                                                                                                                        
     namespace: rook-ceph                                                                                                                     
  spec:                                                                                                                                                   
      metadataPool:                                                                                                                                         
      replicated:                                                                                                                                         
          size: 3                                                                                                                                           
      dataPools:                                                                                                                                            
          - replicated:                                                                                                                                       
                 size: 3                                                                                                                                         
       preservePoolsOnDelete: true                                                                                                                           
       metadataServer:                                                                                                                                       
           activeCount: 1                                                                                                                                      
           activeStandby: true
```

Events from the rook-ceph-crashcollector:
```

Events:                                                                                                                                                 
    Type     Reason       Age                  From                                                 Message                                               
    ----     ------       ----                 ----                                                 -------                                               
    Warning  FailedMount  25m (x4 over 115m)   kubelet, ip-10-1-0-10.eu-central-1.compute.internal  Unable to attach or mount volumes: unmounted volu     
  mes=[rook-ceph-crash-collector-keyring], unattached volumes=[default-token-lc77h rook-ceph-crash-collector-keyring rook-config-override rook-ceph-l     
  og rook-ceph-crash]: timed out waiting for the condition                                                                                                
    Warning  FailedMount  16m (x9 over 120m)   kubelet, ip-10-1-0-10.eu-central-1.compute.internal  Unable to attach or mount volumes: unmounted volu     
  mes=[rook-ceph-crash-collector-keyring], unattached volumes=[rook-ceph-log rook-ceph-crash default-token-lc77h rook-ceph-crash-collector-keyring ro     
  ok-config-override]: timed out waiting for the condition                                                                                                
    Warning  FailedMount  11m (x5 over 109m)   kubelet, ip-10-1-0-10.eu-central-1.compute.internal  Unable to attach or mount volumes: unmounted volu     
  mes=[rook-ceph-crash-collector-keyring], unattached volumes=[rook-ceph-crash default-token-lc77h rook-ceph-crash-collector-keyring rook-config-over     
  ride rook-ceph-log]: timed out waiting for the condition                                                                                                
    Warning  FailedMount  49s (x70 over 127m)  kubelet, ip-10-1-0-10.eu-central-1.compute.internal  MountVolume.SetUp failed for volume "rook-ceph-cr     
  ash-collector-keyring" : secret "rook-ceph-crash-collector-keyring" not found
```

**Comment by BlaineEXE** [comment]
Try making the cluster name and cluster namespace the same and then doing this. Rook expects these to be the same in most cases. Try this and see if the problem persists.

**Comment by harryleesan** [comment]
I have made the cluster name and cluster namespace the same. The issue persists.

rook-operator logs:
```
2020-01-14 12:22:29.737875 I | exec: Running command: ceph version --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-
file /tmp/395768872
2020-01-14 12:22:30.543898 I | exec: Running command: ceph fs get myfs --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --
out-file /tmp/025276775
2020-01-14 12:22:31.413556 I | exec: Error ENOENT: filesystem 'myfs' not found
2020-01-14 12:22:31.413754 I | exec: Running command: ceph fs ls --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-fi
le /tmp/546714010
2020-01-14 12:22:32.318895 I | exec: Running command: ceph osd lspools --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --
out-file /tmp/854756913
2020-01-14 12:22:33.217733 I | op-file: Creating filesystem myfs
2020-01-14 12:22:33.217843 I | exec: Running command: ceph osd crush rule create-replicated myfs-metadata default host --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook
/rook-ceph/client.admin.keyring --format json --out-file /tmp/347641308
2020-01-14 12:22:34.034734 I | exec: rule myfs-metadata already exists
2020-01-14 12:22:34.034988 I | exec: Running command: ceph osd pool create myfs-metadata 0 replicated myfs-metadata --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/ro
ok-ceph/client.admin.keyring --format json --out-file /tmp/221814667
2020-01-14 12:22:35.922080 I | exec: pool 'myfs-metadata' created
2020-01-14 12:22:35.922349 I | exec: Running command: ceph osd pool set myfs-metadata size 3 --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.ke
yring --format json --out-file /tmp/417426286
2020-01-14 12:22:36.931765 I | exec: set pool 3 size to 3
2020-01-14 12:22:36.931873 I | cephclient: creating replicated pool myfs-metadata succeeded, buf:
2020-01-14 12:22:36.931952 I | exec: Running command: ceph osd crush rule create-replicated myfs-data0 default host --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/ro
ok-ceph/client.admin.keyring --format json --out-file /tmp/002798581
2020-01-14 12:22:38.007298 I | exec: Running command: ceph osd pool create myfs-data0 0 replicated myfs-data0 --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-cep
h/client.admin.keyring --format json --out-file /tmp/183603695
2020-01-14 12:22:41.215904 I | exec: pool 'myfs-data0' created
2020-01-14 12:22:41.216449 I | exec: Running command: ceph osd pool set myfs-data0 size 3 --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyri
ng --format json --out-file /tmp/095910530
2020-01-14 12:22:43.247344 I | exec: set pool 4 size to 3
2020-01-14 12:22:43.247512 I | exec: Running command: ceph osd pool application enable myfs-data0 cephfs --yes-i-really-mean-it --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var
/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/762994681
2020-01-14 12:22:44.252045 I | exec: enabled application 'cephfs' on pool 'myfs-data0'
2020-01-14 12:22:44.252261 I | cephclient: creating replicated pool myfs-data0 succeeded, buf:
2020-01-14 12:22:44.252347 I | exec: Running command: ceph fs new myfs myfs-metadata myfs-data0 --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin
.keyring --format json --out-file /tmp/114403076
2020-01-14 12:22:45.103551 I | exec: new fs with metadata pool 3 and data pool 4
2020-01-14 12:22:45.103653 I | op-file: created filesystem myfs on 1 data pool(s) and metadata pool myfs-metadata
2020-01-14 12:22:45.103725 I | exec: Running command: ceph fs get myfs --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --
out-file /tmp/577048211
2020-01-14 12:22:45.928742 I | exec: Running command: ceph fs set myfs allow_standby_replay true --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admi
n.keyring --format json --out-file /tmp/881110742
2020-01-14 12:22:47.328937 I | op-file: start running mdses for filesystem myfs
2020-01-14 12:22:47.329722 I | exec: Running command: ceph auth get-or-create-key mds.myfs-a osd allow * mds allow mon allow profile mds --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --key
ring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/993396797
2020-01-14 12:22:48.320751 I | op-mds: setting mds config flags
2020-01-14 12:22:48.329035 I | exec: Running command: ceph auth get-or-create-key mds.myfs-b osd allow * mds allow mon allow profile mds --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --key
ring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/847275640
2020-01-14 12:22:48.411493 I | op-k8sutil: format and nodeName longer than 63 chars, nodeName ip-10-1-0-12.eu-central-1.compute.internal will be a5122d7224eb942f16318662086c8dae
2020-01-14 12:22:48.427504 I | op-k8sutil: format and nodeName longer than 63 chars, nodeName ip-10-1-0-12.eu-central-1.compute.internal will be a5122d7224eb942f16318662086c8dae
2020-01-14 12:22:48.470580 E | ceph-crashcollector-controller: node reconcile failed on op "unchanged": Operation cannot be fulfilled on deployments.apps "rook-ceph-crashcollector-a5122d7224eb942f16318662086c8dae": the object has b
een modified; please apply your changes to the latest version and try again
2020-01-14 12:22:49.408446 I | op-mds: setting mds config flags
2020-01-14 12:22:49.473105 I | op-k8sutil: format and nodeName longer than 63 chars, nodeName ip-10-1-0-12.eu-central-1.compute.internal will be a5122d7224eb942f16318662086c8dae
2020-01-14 12:22:49.479671 I | op-k8sutil: format and nodeName longer than 63 chars, nodeName ip-10-1-0-5.eu-central-1.compute.internal will be e0642b709f67afd086bb87dff980ec03
2020-01-14 12:22:49.485821 I | op-k8sutil: format and nodeName longer than 63 chars, nodeName ip-10-1-0-5.eu-central-1.compute.internal will be e0642b709f67afd086bb87dff980ec03
2020-01-14 12:22:49.510606 I | op-k8sutil: format and nodeName longer than 63 chars, nodeName ip-10-1-0-5.eu-central-1.compute.internal will be e0642b709f67afd086bb87dff980ec03
2020-01-14 12:22:49.513779 E | ceph-crashcollector-controller: node reconcile failed on op "unchanged": Operation cannot be fulfilled on deployments.apps "rook-ceph-crashcollector-e0642b709f67afd086bb87dff980ec03": the object has b
een modified; please apply your changes to the latest version and try again
2020-01-14 12:22:50.516473 I | op-k8sutil: format and nodeName longer than 63 chars, nodeName ip-10-1-0-5.eu-central-1.compute.internal will be e0642b709f67afd086bb87dff980ec03
2020-01-14 12:23:10.823662 I | op-mon: ClusterInfo is now Empty, refilling it from status.MonMap.Mons
2020-01-14 12:23:10.823687 I | op-mon: new external mon ip-10-1-0-24 found: 10.1.0.24:6789, adding it
2020-01-14 12:23:10.823695 I | op-mon: new external mon ip-10-1-0-25 found: 10.1.0.25:6789, adding it
2020-01-14 12:23:10.823702 I | op-mon: new external mon ip-10-1-0-27 found: 10.1.0.27:6789, adding it
2020-01-14 12:23:10.830151 I | op-mon: saved mon endpoints to config map map[csi-cluster-config-json:[{"clusterID":"rook-ceph","monitors":["10.1.0.27:6789","10.1.0.24:6789","10.1.0.25:6789"]}] data:ip-10-1-0-24=10.1.0.24:6789,ip-10
-1-0-25=10.1.0.25:6789,ip-10-1-0-27=10.1.0.27:6789 maxMonId:-1 mapping:{"node":{}}]
2020-01-14 12:23:10.833728 I | cephconfig: writing config file /var/lib/rook/rook-ceph/rook-ceph.config
2020-01-14 12:23:10.833828 I | cephconfig: generated admin config in /var/lib/rook/rook-ceph
```

`kubectl get po`:
```
csi-cephfsplugin-77wkq                                            3/3     Running    0          3h22m
csi-cephfsplugin-provisioner-56c8b7ddf4-f7qst                     4/4     Running    0          3h22m
csi-cephfsplugin-provisioner-56c8b7ddf4-pszk8                     4/4     Running    0          3h22m
csi-cephfsplugin-w7sff                                            3/3     Running    0          3h22m
csi-cephfsplugin-xbxxn                                            3/3     Running    0          3h22m
csi-rbdplugin-25jc7                                               3/3     Running    0          3h22m
csi-rbdplugin-62wlr                                               3/3     Running    0          3h22m
csi-rbdplugin-gqh6f                                               3/3     Running    0          3h22m
csi-rbdplugin-provisioner-6ff4dd4b94-d5wts                        5/5     Running    0          3h22m
csi-rbdplugin-provisioner-6ff4dd4b94-dg8x5                        5/5     Running    0          3h22m
rook-ceph-crashcollector-a5122d7224eb942f16318662086c8dae-kcgq7   0/1     Init:0/2   0          63s
rook-ceph-crashcollector-e0642b709f67afd086bb87dff980ec03-dqjzf   0/1     Init:0/2   0          62s
rook-ceph-mds-myfs-a-5cb944748f-tzh5t                             0/1     Init:0/1   0          63s
rook-ceph-mds-myfs-b-7bbd558546-nrxs5                             0/1     Init:0/1   0          62s
rook-ceph-operator-dbfd9697b-bbsvc                                1/1     Running    0          3h35m
rook-discover-887k6                                               1/1     Running    0          3h34m
rook-discover-b78h6                                               1/1     Running    0          3h34m
rook-discover-wws4w                                               1/1     Running    0          3h34m
```

`kubectl describe po rook-ceph-mds-myfs-a-5cb944748f-tzh5t`:
```
...
Events:
  Type     Reason       Age                 From                                                 Message
  ----     ------       ----                ----                                                 -------
  Normal   Scheduled    <unknown>           default-scheduler                                    Successfully assigned rook-ceph/rook-ceph-mds-myfs-a-5cb944748f-tzh5t to ip-10-1-0-12.eu-central-1.compute.internal
  Warning  FailedMount  61s                 kubelet, ip-10-1-0-12.eu-central-1.compute.internal  Unable to attach or mount volumes: unmounted volumes=[rook-config-override], unattached volumes=[rook-config-override rook-ceph-mds-m
fs-a-keyring rook-ceph-log rook-ceph-crash ceph-daemon-data default-token-shdj9]: timed out waiting for the condition
  Warning  FailedMount  56s (x9 over 3m4s)  kubelet, ip-10-1-0-12.eu-central-1.compute.internal  MountVolume.SetUp failed for volume "rook-config-override" : configmap "rook-config-override" not found
```

After I have created the `rook-config-override` configmap, the `mds` pods are able to start up.

**Comment by BlaineEXE** [comment]
The operator should create the configmap on startup for both external and non-external clusters. I'm not sure why it isn't. A link to a pastebin of the full operator logs are probably needed.

**Comment by leseb** [comment]
Sorry for the late reply, https://github.com/rook/rook/pull/4688 will fix your issue @harryleesan

**Comment by toydarian** [comment]
I encounter this issue with a local (non-external) cluster with rook on master and the latest master image from ceph.  
Disabling the crash-collector works as a workaround. Is it planned to fix the issue?

**Comment by leseb** [comment]
@toydarian did you wait enough time? Sometimes it can take a couple of seconds but eventually, the secret gets created and the crash pod starts.

**Comment by toydarian** [comment]
After over 20 minutes the container still hasn't started:
```
Events:
  Type     Reason       Age                   From               Message
  ----     ------       ----                  ----               -------
  Normal   Scheduled    <unknown>             default-scheduler  Successfully assigned rook-ceph/rook-ceph-crashcollector-benni-5d7bf9689f-n6cw8 to benni
  Warning  FailedMount  20m (x2 over 22m)     kubelet, benni     Unable to attach or mount volumes: unmounted volumes=[rook-ceph-crash-collector-keyring], unattached volumes=[rook-ceph-crash-collector-keyring rook-config-override rook-ceph-log rook-ceph-crash default-token-gwrjm]: timed out waiting for the condition
```
I'm using rook master and the latest ceph image: `ceph/daemon-base:latest-master`

**Comment by nielsbasjes** [comment]
FYI: I now have everything running.

From what I understand now I did two things "wrong" (I may be wrong in this analysis, but I'm sharing it anyway with you guys)

1. I was too impatient. I did the the steps one by one and waited until everything was finished before I did the next step. Some things "look green" but are really jobs that need to complete before you can proceed. 
2. I switched from flannel to calico. I had many problems regarding networking and the simple switch to calico solved all of them. Also the problems I had with rook were "all gone" after this change.

Note: I installed my cluster using this puppet module: https://forge.puppet.com/puppetlabs/kubernetes

So from my standpoint it seems that some "rook-ceph" problems are really cluster problems.
I think a (link to a) basic step-by-step checklist to verify that the basic networking is in order on the troubleshooting page would be nice.

**Comment by toydarian** [comment]
I fixed my problem. Description what went wrong in my very specific setup is in this comment: https://github.com/rook/rook/issues/4715#issuecomment-580770591

**Comment by kevinsingapore** [comment]
> I think I figured it out. Writing this down here for others to find.
> I had done several tests installing rook ceph in Kubernetes.
> Cleaning a system for a new run takes more than just removing the /var/lib/rook directory.
> I also found some sockets in the /var/lib/kubelet directory related to rook ceph
> On my system it worked after removing these too.

this method works ok!
the tree related path:
(1)./var/lib/rook/
(2)./var/lib/kubelet/plugins/
(3)./var/lib/kubelet/plugins_registry/
afer u clear the files, remember to reinstall the rook service, otherwise it doesn't work.

**Comment by JieZeng1993** [comment]
> This bug seems to appear in Rook 1.2 when connecting to an **external** Ceph cluster.
> 
> After creating the CRs: `CephCluster` (external=true) -> `CephFilesystem`, the `rook-ceph-crash-collector-keyring` secret is not created and crash reporter fails to start.
> 
> I had to create the rook-ceph-crash-collector-keyring manually.

could you tell how create rook-ceph-crash-collector-keyring  manually?
i don`t known how to do it

**Comment by alter** [comment]
@JieZeng1993 `kubectl -n rook-ceph create secret generic rook-ceph-crash-collector-keyring`

**Comment by avishaiNB** [comment]
@alter this is not always the case, sometimes it is just a symptom.

Common use case is when you deploy the cluster and for some reason you remove it without properly purging the servers (**dataDirHostPath** folder).

I had a mix problem when i used your solution to solve the secret issue, but this solution also "blocked" the mgr pod from even getting created (something with "...field is immutable.")

When i removed the secret (and restarted the operator pod), everything started to work as expected.

The initial problem was purging the **dataDirHostPath** folder on each server.

Hope it helps anybody :)

**Comment by Hywfred** [comment]
> kubectl -n rook-ceph create secret generic rook-ceph-crash-collector-keyring

That works for me! Thanks man!

**Comment by kubealex** [comment]
Still present in 1.5.3

**Comment by minhnnhat** [comment]
> @alter this is not always the case, sometimes it is just a symptom.
> 
> Common use case is when you deploy the cluster and for some reason you remove it without properly purging the servers (**dataDirHostPath** folder).
> 
> I had a mix problem when i used your solution to solve the secret issue, but this solution also "blocked" the mgr pod from even getting created (something with "...field is immutable.")
> 
> When i removed the secret (and restarted the operator pod), everything started to work as expected.
> 
> The initial problem was purging the **dataDirHostPath** folder on each server.
> 
> Hope it helps anybody :)

It's perfectly work, thank you so much !

**Comment by sd-hd** [comment]
> @JieZeng1993 `kubectl -n rook-ceph create secret generic rook-ceph-crash-collector-keyring`

I had to use this:

kubectl -n rook-ceph create secret generic --type kubernetes.io/rook rook-ceph-crash-collector-keyring

**Comment by vishalk663** [comment]
> I think I figured it out. Writing this down here for others to find. I had done several tests installing rook ceph in Kubernetes. Cleaning a system for a new run takes more than just removing the /var/lib/rook directory. I also found some sockets in the /var/lib/kubelet directory related to rook ceph On my system it worked after removing these too.

I am also facing this error. after reinstalling also we are getting the same error.
We zap the disk and remove the socket files as well related to ceph from /var/lib/kubelet/puugins and /var/lib/kubelet/plugins_registry directory

**Comment by travisn** [comment]
@vishalk663 Want to open a new issue (or even better a PR) so we can get this documented in the [cleanup guide](https://github.com/rook/rook/blob/master/Documentation/ceph-teardown.md)?

**Comment by huangxiaofeng10047** [comment]
> > @JieZeng1993 `kubectl -n rook-ceph create secret generic rook-ceph-crash-collector-keyring`
> 
> I had to use this:
> 
> kubectl -n rook-ceph create secret generic --type kubernetes.io/rook rook-ceph-crash-collector-keyring

I use this command ，then it worked.
